__tests__/dockerhub.test.ts

```diff
@@ -1,121 +0,0 @@
-
-beforeEach(() => {
-  jest.clearAllMocks();
-});
-
-import repoInfoFixture from './fixtures/dockerhub-repoinfo.json';
-import repoTagsFixture from './fixtures/dockerhub-repotags.json';
-import repoAllTagsFixture from './fixtures/dockerhub-repoalltags.json';
-
-describe('getRepository', () => {
-  it('returns repo info', async () => {
-    jest.spyOn(DockerHub.prototype, 'getRepository').mockImplementation((): Promise<RepositoryResponse> => {
-      return <Promise<RepositoryResponse>>(repoInfoFixture as unknown);
-    });
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    jest.spyOn(DockerHub as any, 'login').mockReturnValue('jwt_token');
-    const dockerhub = await DockerHub.build({
-      credentials: {
-        username: 'foo',
-        password: '0123456-7890-0000-1111-222222222'
-      }
-    });
-    const repoinfo = await dockerhub.getRepository({
-      namespace: 'foo',
-      name: 'bar'
-    });
-    expect(repoinfo.namespace).toEqual('foo');
-    expect(repoinfo.name).toEqual('bar');
-    expect(repoinfo.repository_type).toEqual('image');
-  });
-});
-
-describe('getRepositoryTags', () => {
-  it('return repo tags', async () => {
-    jest.spyOn(DockerHub.prototype, 'getRepositoryTags').mockImplementation((): Promise<RepositoryTagsResponse> => {
-      return <Promise<RepositoryTagsResponse>>(repoTagsFixture as unknown);
-    });
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    jest.spyOn(DockerHub as any, 'login').mockReturnValue('jwt_token');
-    const dockerhub = await DockerHub.build({
-      credentials: {
-        username: 'foo',
-        password: '0123456-7890-0000-1111-222222222'
-      }
-    });
-    const resp = await dockerhub.getRepositoryTags({
-      namespace: 'crazymax',
-      name: 'diun'
-    });
-    expect(resp.count).toBeGreaterThan(0);
-    expect(resp.next).not.toBeNull();
-    expect(resp.results.length).toBeGreaterThan(0);
-    expect(resp.results[0].last_updater_username).toEqual('crazymax');
-  });
-});
-
-describe('getRepositoryAllTags', () => {
-  it('return repo all tags', async () => {
-    jest.spyOn(DockerHub.prototype, 'getRepositoryAllTags').mockImplementation((): Promise<RepositoryTagsResponse> => {
-      return <Promise<RepositoryTagsResponse>>(repoAllTagsFixture as unknown);
-    });
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    jest.spyOn(DockerHub as any, 'login').mockReturnValue('jwt_token');
-    const dockerhub = await DockerHub.build({
-      credentials: {
-        username: 'foo',
-        password: '0123456-7890-0000-1111-222222222'
-      }
-    });
-    const resp = await dockerhub.getRepositoryAllTags({
-      namespace: 'crazymax',
-      name: 'diun'
-    });
-    expect(resp.count).toBeGreaterThan(0);
-    expect(resp.next).toBeNull();
-    expect(resp.results.length).toBeGreaterThan(0);
-    expect(resp.results[0].last_updater_username).toEqual('crazymax');
-  });
```

Comment:
This test seems to be dropped now

Reply:
Yeah, I removed the function all together. Per my comment in the description, I don't think we should provide this functionality out of the box. It should be up to the implementation to decide if they want to endlessly run out of API calls 😓.